src/main/scala/algolia/definitions/WaitForTaskDefinition.scala

```diff
@@ -64,6 +60,19 @@ trait WaitForTaskDsl {
   implicit object WaitForTaskDefinitionExecutable
       extends Executable[WaitForTaskDefinition, TaskStatus] {
 
+    // Run every 50 ms, use a wheel with 512 buckets
+    private lazy val timer = {
+      val threadFactory = new ThreadFactory {
```

Comment:
Yes, you are right the current implementation was using an internal pool of threads, but I was not very big fan of it.
At least, could you expose the number of threads of this pool in the configuration? So people could tweak it.